api/rds.proto

```diff
@@ -797,3 +805,22 @@ message HeaderMatcher {
   // * The regex *\d{3}* does not match the value *123.456*
   google.protobuf.BoolValue regex = 3;
 }
+
+// Query arg matching treats the query string of a request's :path header
+// as an ampersand-separated list of keys and/or key=value elements.
+message QueryArgMatcher {
```

Comment:
Consider embedding in `RouteMatch`, it will render in docs closer to its use and is a logical grouping.

Reply:
Do you anticipate `QueryArgMatcher` will be useful in places outside `RouteMatch`, like how `HeaderMatcher` is used in both `RouteMatch` and `RateLimit`?

Reply:
That's a good point. We may make use if it elsewhere in rate limiting, etc. I could go either way, will defer to @htuch.